### PR TITLE
Create API public/private methods

### LINKED ISSUE #118: Create api methods as example
One public api method
one private api method
test both

Note:  the first person who comments on this issue can start working on it, the maintainers will assign you after that, you don't need to wait for approval to start coding

### PR DESCRIPTION
**Pull request by philipeleandro**
Fix #118 

- Add private and public API methods with tests

### CODE REVIEW

File: spec/requests/api/hello_word_spec.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+      it "receives http status" do
+        expect(response).to have_http_status(:unauthorized)
+      end
+    end
+
+    context "when user is authenticated" do
+      before do
+        post "/users/tokens/sign_in", params: { email: user.email, password: user.password }, as: :json
+        json_response = response.parsed_body
```
Comment (edimossilva) [inline spec/requests/api/hello_word_spec.rb:38]: I think there is a way to generate this without doing a post request, using a helper class or something. Can you double check and let me know?

If there is no way of generating it using the library, we need to create our own so we can reuse it for other methods

**Review by edimossilva — COMMENTED** [review]
Looks good to me, just a small comment about improving a pec

File: spec/requests/api/hello_word_spec.rb (hunk 1 above)
Comment (philipeleandro) [reply spec/requests/api/hello_word_spec.rb]: Thanks for review it. I searched on devise-api's issues and I found a comment about this. Can you review again? @edimossilva

Comment (philipeleandro) [reply spec/requests/api/hello_word_spec.rb]: I don't know if it's the better way to use it with a `let`. Is it better to set it up in a helper?

File: spec/requests/api/hello_word_spec.rb [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+RSpec.describe "HelloWorlds" do
+  describe "GET /public_method" do
+    before { get "/api/public_method" }
+
+    it "renders json with message" do
+      expect(response.body).to include("This method does not need authentication")
+    end
+
+    it "receives http status" do
```
Comment (edimossilva) [inline spec/requests/api/hello_word_spec.rb:13]:
```suggestion
    it "receives http status ok" do
```

File: spec/requests/api/hello_word_spec.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+      before do
+        instance_token.call
+        get "/api/private_method", params: {}, headers: { Authorization: token }
+      end
+
+      it "renders json with message" do
+        expect(response.body).to include("This method needs authentication")
+      end
+
+      it "receives http status" do
```
Comment (edimossilva) [inline spec/requests/api/hello_word_spec.rb:48]:
```suggestion
      it "receives http status ok" do
```

File: spec/requests/api/hello_word_spec.rb [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    context "when user is not authenticated" do
+      before do
+        get "/api/private_method"
+      end
+
+      it "renders json with message" do
+        expect(response.body).to include("error", "Invalid token")
+      end
+
+      it "receives http status" do
```
Comment (edimossilva) [inline spec/requests/api/hello_word_spec.rb:30]:
```suggestion
      it "receives http status unauthorized" do
```

Comment (philipeleandro) [reply spec/requests/api/hello_word_spec.rb]: changed!

File: spec/requests/api/hello_word_spec.rb (hunk 3 above)
Comment (philipeleandro) [reply spec/requests/api/hello_word_spec.rb]: changed!

**Review by edimossilva — APPROVED** [review]
Only the filet!

**Pull request merged**